dash/dependencies.py

```diff
@@ -1,19 +1,35 @@
-# pylint: disable=old-style-class, too-few-public-methods
-class Output:
+# pylint: disable=old-style-class
+class DashDependency:
     def __init__(self, component_id, component_property):
         self.component_id = component_id
         self.component_property = component_property
 
+    def __str__(self):
+        return '{}.{}'.format(
+            self.component_id,
+            self.component_property
+        )
 
-# pylint: disable=old-style-class, too-few-public-methods
-class Input:
-    def __init__(self, component_id, component_property):
-        self.component_id = component_id
-        self.component_property = component_property
+    def __repr__(self):
+        return '<{} `{}`>'.format(self.__class__.__name__, self)
 
+    def __eq__(self, other):
+        return str(self) == str(other)
```

Comment:
Do we want to restrict this to `isinstance(other, DashDependency)`? As is, `Input('id', 'value') == 'id.value'` which feels weird to me, but within our own classes it should be fine.